password/test/tasty/Validate.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE CPP             #-}
+{-# LANGUAGE LambdaCase      #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Validate where
+
+import Control.Monad (replicateM)
+import Data.Char (chr, isAsciiLower, isAsciiUpper, isControl, isDigit)
+import Data.Maybe (catMaybes, fromMaybe)
+import Data.Password (mkPassword)
+import Data.Password.Validate (CharSetPredicate (..), CharacterCategory (..),
+                               InvalidReason (..), PasswordPolicy (..),
+                               defaultCharSet, defaultCharSetPredicate,
+                               isSpecial, isValidPasswordPolicy,
+                               validatePassword)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Test.QuickCheck.Instances.Text ()
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.QuickCheck (Arbitrary (..), Gen, Property, choose, conjoin,
+                              elements, liftArbitrary, oneof, shuffle, suchThat,
+                              testProperty, withMaxSuccess, (===))
+#if! MIN_VERSION_base(4,13,0)
+import Data.Semigroup ((<>))
+#endif
+
+-- | Set of tests used for testing validate module
+testValidate :: TestTree
+testValidate =
+  testGroup
+    "validate"
+    [ testGroup "defaultCharSetPredicate"
+        [ testProperty "Return false on all control characters"
+            (not $ all (getCharSetPredicate defaultCharSetPredicate) asciiControlChars),
+          testProperty "Return true on all non-control characters"
+            (all (getCharSetPredicate defaultCharSetPredicate) asciiNonControlChars)
+        ]
+      ,
+      testProperty "defaultCharSet should contain all the expected characters" $
+        (conjoin
+          [ hasCharacters isAsciiUpper 26
+          , hasCharacters isAsciiLower 26
+          , hasCharacters isDigit 10
+          , hasCharacters isSpecial 33
+          ]
+        ),
```

Comment:
Added test which tests that the `defaultCharSet` contains all the expected characters.